File: spark/src/main/scala/com/adatao/ddf/spark/DDFDriver.scala
package com.adatao.ddf.spark

import com.adatao.ddf.DDFDriver
import com.adatao.ddf.DDF
import com.adatao.ddf.DDFDriverManager
import com.adatao.ddf.exception.DDFException

class DDFDriver extends com.adatao.ddf.DDFDriver {
  DDFDriver // to trigger the static call to register the driver with driver manager.
  
  def createSchema(schemaString: java.lang.String): DDF = {
    return null
  }

  def fromSql(sqlCommand: java.lang.String): DDF = {
    return null
  }
  
  def acceptURL(connectionURL: java.lang.String): Boolean = {
    return true
  }
}

object DDFDriver {
  try {
    DDFDriverManager.registerDDFDriver(new com.adatao.ddf.spark.DDFDriver)
  } catch {
    case ex: DDFException => println(ex.getMessage)
  }
}

File: spark/src/main/java/com/adatao/ddf/spark/DDFDriver.java
package com.adatao.ddf.spark;

import java.util.Map;

import org.apache.spark.api.java.JavaSparkContext;

import com.adatao.ddf.DDFContextManager;
import com.adatao.ddf.DDFFactory;
import com.adatao.ddf.exception.DDFException;

public class DDFContext implements com.adatao.ddf.DDFContext {
    static {
        try {
            DDFContextManager.registerDDFContext(new DDFContext());
        } catch (DDFException e) {
            throw new RuntimeException("Cannot register the Spark DDF Context.");
        }
    }
    
    @Override
    public boolean acceptURL(String connectionURL) throws DDFException {
        return true;
    }

    @Override
    public DDFFactory connect(String connectionURL,
            Map<String, String> connectionProps) throws DDFException {
        String[] jobJars = connectionProps.get("DDFSPARK_JAR").split(",");
        JavaSparkContext sc = new JavaSparkContext(connectionURL, "DDFClient", connectionProps.get("SPARK_HOME"), jobJars, connectionProps);
        return new SparkDDFFactory(sc);
    }

}
